donkeycar/parts/controller.py

```diff
@@ -227,107 +228,138 @@ def poll(self):
-            t = pigpio.tickDiff(self.high_tick, tick)
-            return t
-
-    def _cbf(self, gpio, level, tick):
         """ Callback function for pigpio interrupt gpio. Signature is determined
             by pigpiod library. This function is called every time the gpio
-            changes state as we specified EITHER_EDGE.
+            changes state as we specified EITHER_EDGE.  The pigpio callback library
+            sends the user-defined callback function three parameters, which it may or may not use
         :param gpio: gpio to listen for state changes
         :param level: rising/falling edge
         :param tick: # of mu s since boot, 32 bit int
         """
-        if level == 1:
-            self.period = self._update_param(tick)
-            self.high_tick = tick
-        elif level == 0:
-            self.high = self._update_param(tick)
-
-    def pulse_width(self):
+        if gpio == self.RC1_in_PIN:
+            if level == 1:
+                self.high_tick1 = tick
+            elif level == 0:
+                if self.high_tick1 is not None:
+                    self.high1 = pigpio.tickDiff(self.high_tick1, tick)
+        if gpio == self.RC2_in_PIN:
+            if level == 1:
+                self.high_tick2 = tick
+            elif level == 0:
+                if self.high_tick2 is not None:
+                    self.high2 = pigpio.tickDiff(self.high_tick2, tick)
+        if gpio == self.RC3_in_PIN:
+            if level == 1:
+                self.high_tick3 = tick
+            elif level == 0:
+                if self.high_tick3 is not None:
+                    self.high3 = pigpio.tickDiff(self.high_tick3, tick)
+
+
+    def pulse_width(self, high):
         """
         :return: the PWM pulse width in microseconds.
         """
-        if self.high is not None:
-            return self.high
+        if high is not None:
+            return high
         else:
             return 0.0
 
-    def run(self):
+    def run(self, img_arr=None):
         """
         Donkey parts interface, returns pulse mapped into [MIN_OUT,MAX_OUT] or
         [MAX_OUT,MIN_OUT]
         """
         # signal is a value in [0, (MAX_OUT-MIN_OUT)]
-        signal = (self.pulse_width() - self.min_pwm) * self.factor
+        signal1 = (self.pulse_width(self.high1) - self.min_pwm) * self.factor
+        # convert into min max interval
+        if self.invert:
+            signal1 = -signal1 + self.MAX_OUT
+        else:
+            signal1 += self.MIN_OUT
+
+        signal2 = (self.pulse_width(self.high2) - self.min_pwm) * self.factor
         # Assuming non-activity if the pulse is at no_action point
-        is_action = abs(signal - self.no_action) > self.jitter
-        # if deemed noise assume no signal
-        if not is_action:
-            signal = self.no_action
+        if abs(signal2 - self.jitter) > 1.0:  # this is structured as a boolean
+            is_action = True
+        else:
+            is_action = False
```

Comment:
These four lines are simply: `is_action = abs(signal2 - self.jitter) > 1.0`

Reply:
Yes, but in a subsequent commit (sorry again for not squashing!) I added another condition to avoid turning on recording when not wanted. Now that it's a bit more complex, I think it's okay to spell it out a bit more 
` if (abs(signal2 - self.jitter) > 1.0) and self.RECORD: # is throttle above jitter level? If so, turn on auto-record `